tf2_py/src/tf2_py.cpp

```diff
@@ -129,14 +129,30 @@ static PyObject *transform_converter(const geometry_msgs::TransformStamped* tran
   return pinst;
 }
 
-static int rostime_converter(PyObject *obj, builtin_interfaces::msg::Time *rt)
+static int rostime_converter(PyObject *obj, tf2::TimePoint *rt)
 {
   PyObject *tsr = PyObject_CallMethod(obj, (char*)"to_sec", NULL);
   if (tsr == NULL) {
     PyErr_SetString(PyExc_TypeError, "time must have a to_sec method, e.g. rospy.Time or rospy.Duration");
     return 0;
   } else {
-    (*rt).fromSec(PyFloat_AsDouble(tsr));
+    *rt = tf2::timeFromSec(PyFloat_AsDouble(tsr));
+    Py_DECREF(tsr);
+    return 1;
+  }
+}
+
+static int rostime_converter_blt(PyObject *obj, builtin_interfaces::msg::Time *rt)
```

Comment:
We already have an interface to convert from builtin_interfaces::msg::Time -> tf2::TimePoint: https://github.com/ros2/geometry2/blob/ros2/tf2_ros/include/tf2_ros/buffer_interface.h#L58 .  I think it would be best to reuse that, as having the conversions scattered around seems to be more fragile to me.  That being said, that would introduce a new dependency on tf2_py, namely tf2_ros; I'm not sure if that is a problem or not.